### ISSUE TITLE
Sass is rebuilding when any file is changed

### ISSUE THREAD

**Issue by mshindal** [aexmachina/ember-cli-sass#108]
Hi, I'm having an issue where `ember-cli-sass` will rebuild when any file is changed in my project, instead of only rebuilding when an `.scss` file is changed. 

I'm using Bootstrap-Sass and Bootswatch, and I'm installing both of them via Bower. I add them to the include path in my `ember-cli-build.js`:
```
sassOptions: {
      includePaths: [
        'bower_components/bootstrap-sass/assets/stylesheets/',
        'bower_components/bootswatch/paper/'
      ]
    }
```

Then in my `app.scss` I import the files I need:
```
@import 'variables';
@import 'bootstrap';
@import 'bootswatch';

// custom SCSS below...
```

This produces the output that I want, but it rebuilds when ANY file is changed, making `ember server` very slow:
```
file changed templates\index.hbs

Build successful - 7489ms.

Slowest Trees                                 | Total
----------------------------------------------+---------------------
SassCompiler                                  | 6636ms

Slowest Trees (cumulative)                    | Total (avg)
----------------------------------------------+---------------------
SassCompiler (1)                              | 6636ms
```

Any ideas? This is happening with `ember-cli-less` for me as well.

**Comment by stefanpenner** [comment]
i suspect you have some outdated broccoli-plugin somewhere or one that is misbehaving, Please be sure they are all up to date.

**Comment by mshindal** [comment]
What would be a good way to check that? Here's my `npm outdated`:
```
Package                         Current  Wanted   Latest         Location                                                                                                                    
ember-ajax                     0.7.1            0.7.1   2.0.0-beta.2  ember-ajax                                                                                                                 
ember-cli-release              0.2.8            0.2.8   1.0.0-beta.1  ember-cli-release                                                                                                          
ember-simple-auth              1.0.1            1.0.1   1.1.0-beta.3  ember-simple-auth                                                                                                          
ember-cli                      2.3.0            2.3.0   2.4.0         ember-cli                                                                                                                  
is-git-url                     0.2.0            0.2.0   0.2.3         ember-cli-dependency-checker > is-git-url                                                                                  
semver                         4.3.6            4.3.6   5.1.0         ember-cli-dependency-checker > semver                                                                                      
chalk                          0.5.1            0.5.1   1.1.1         ember-cli-dependency-checker > chalk                                                                                       
semver                         4.3.6            4.3.6   5.1.0         ember-resolver > ember-cli-version-checker > semver                                                                        
lodash                         3.10.1           3.10.1  4.5.1         ember-cli-uglify > broccoli-uglify-sourcemap > lodash                                                                      
uglify-js                      2.5.0            2.5.0   2.6.2         ember-cli-uglify > broccoli-uglify-sourcemap > uglify-js                                                                   
source-map-url                 0.3.0            0.3.0   0.4.0         ember-cli-uglify > broccoli-uglify-sourcemap > source-map-url                                                              
walk-sync                      0.1.3            0.1.3   0.2.6         ember-cli-uglify > broccoli-uglify-sourcemap > walk-sync                                                                   
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-uglify > broccoli-uglify-sourcemap > matcher-collection > minimatch                                              
minimist                       0.0.8            0.0.8   1.2.0         ember-cli-uglify > broccoli-uglify-sourcemap > mkdirp > minimist                                                           
rimraf                         2.2.8            2.2.8   2.5.2         ember-cli-uglify > broccoli-uglify-sourcemap > broccoli-writer > quick-temp > rimraf                                       
underscore.string              2.3.3            2.3.3   3.3.4         ember-cli-uglify > broccoli-uglify-sourcemap > broccoli-writer > quick-temp > underscore.string                            
mktemp                         0.3.5            0.3.5   0.4.0         ember-cli-uglify > broccoli-uglify-sourcemap > broccoli-writer > quick-temp > mktemp                                       
minimist                       0.0.8            0.0.8   1.2.0         ember-cli-sri > broccoli-sri-hash > mkdirp > minimist                                                                      
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-cli-sri > broccoli-sri-hash > broccoli-caching-writer > broccoli-kitchen-sink-helpers                                
broccoli-plugin                1.1.0            1.1.0   1.2.1         ember-cli-sri > broccoli-sri-hash > broccoli-caching-writer > broccoli-plugin                                              
lodash.assign                  3.2.0            3.2.0   4.0.4         ember-cli-sri > broccoli-sri-hash > broccoli-caching-writer > lodash.assign                                                
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-sri > broccoli-sri-hash > broccoli-caching-writer > walk-sync > matcher-collection > minimatch                   
semver                         4.3.6            4.3.6   5.1.0         ember-cli-htmlbars > ember-cli-version-checker > semver                                                                    
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-cli-htmlbars > broccoli-persistent-filter > broccoli-kitchen-sink-helpers                                            
fs-tree-diff                   0.3.1            0.3.1   0.4.4         ember-cli-htmlbars > broccoli-persistent-filter > fs-tree-diff                                                             
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-cli-htmlbars > broccoli-persistent-filter > hash-for-dep > broccoli-kitchen-sink-helpers                             
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-htmlbars > broccoli-persistent-filter > walk-sync > matcher-collection > minimatch                               
minimist                       0.0.8            0.0.8   1.2.0         ember-cli-htmlbars > broccoli-persistent-filter > mkdirp > minimist                                                        
rimraf                         2.2.8            2.2.8   2.5.2         ember-cli-htmlbars > broccoli-persistent-filter > broccoli-plugin > quick-temp > rimraf                                    
mktemp                         0.3.5            0.3.5   0.4.0         ember-cli-htmlbars > broccoli-persistent-filter > broccoli-plugin > quick-temp > mktemp                                    
underscore.string              2.3.3            2.3.3   3.3.4         ember-cli-htmlbars > broccoli-persistent-filter > broccoli-plugin > quick-temp > underscore.string                         
semver                         4.3.6            4.3.6   5.1.0         ember-cli-babel > ember-cli-version-checker > semver                                                                       
fs-tree-diff                   0.3.1            0.3.1   0.4.4         ember-cli-babel > broccoli-funnel > fs-tree-diff                                                                           
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-babel > broccoli-funnel > minimatch                                                                              
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-babel > broccoli-funnel > walk-sync > matcher-collection > minimatch                                             
minimist                       0.0.8            0.0.8   1.2.0         ember-cli-babel > broccoli-funnel > mkdirp > minimist                                                                      
rimraf                         2.2.8            2.2.8   2.5.2         ember-cli-babel > broccoli-funnel > broccoli-plugin > quick-temp > rimraf                                                  
underscore.string              2.3.3            2.3.3   3.3.4         ember-cli-babel > broccoli-funnel > broccoli-plugin > quick-temp > underscore.string                                       
mktemp                         0.3.5            0.3.5   0.4.0         ember-cli-babel > broccoli-funnel > broccoli-plugin > quick-temp > mktemp                                                  
minimatch                      2.0.10           3.0.0   3.0.0         ember-cli-babel > broccoli-funnel > rimraf > glob > minimatch                                                              
babel-core                     5.8.35           5.8.35  6.6.0         ember-cli-babel > broccoli-babel-transpiler > babel-core                                                                   
clone                          0.2.0            0.2.0   1.0.2         ember-cli-babel > broccoli-babel-transpiler > clone                                                                        
rimraf                         2.2.8            2.2.8   2.5.2         ember-cli-babel > broccoli-babel-transpiler > broccoli-merge-trees > broccoli-plugin > quick-temp > rimraf                 
mktemp                         0.3.5            0.3.5   0.4.0         ember-cli-babel > broccoli-babel-transpiler > broccoli-merge-trees > broccoli-plugin > quick-temp > mktemp                 
underscore.string              2.3.3            2.3.3   3.3.4         ember-cli-babel > broccoli-babel-transpiler > broccoli-merge-trees > broccoli-plugin > quick-temp > underscore.string      
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-cli-babel > broccoli-babel-transpiler > broccoli-persistent-filter > broccoli-kitchen-sink-helpers                   
fs-tree-diff                   0.3.1            0.3.1   0.4.4         ember-cli-babel > broccoli-babel-transpiler > broccoli-persistent-filter > fs-tree-diff                                    
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-cli-babel > broccoli-babel-transpiler > broccoli-persistent-filter > hash-for-dep > broccoli-kitchen-sink-helpers    
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-babel > broccoli-babel-transpiler > broccoli-persistent-filter > walk-sync > matcher-collection > minimatch      
minimist                       0.0.8            0.0.8   1.2.0         ember-cli-babel > broccoli-babel-transpiler > broccoli-persistent-filter > mkdirp > minimist                               
rimraf                         2.2.8            2.2.8   2.5.2         ember-cli-babel > broccoli-babel-transpiler > broccoli-persistent-filter > broccoli-plugin > quick-temp > rimraf           
mktemp                         0.3.5            0.3.5   0.4.0         ember-cli-babel > broccoli-babel-transpiler > broccoli-persistent-filter > broccoli-plugin > quick-temp > mktemp           
underscore.string              2.3.3            2.3.3   3.3.4         ember-cli-babel > broccoli-babel-transpiler > broccoli-persistent-filter > broccoli-plugin > quick-temp > underscore.string
semver                         4.3.6            4.3.6   5.1.0         ember-data > ember-cli-version-checker > semver                                                                            
supports-color                 2.0.0            2.0.0   3.1.2         ember-data > chalk > supports-color                                                                                        
clone                          0.2.0            0.2.0   1.0.2         ember-data > broccoli-babel-transpiler > clone                                                                             
babel-core                     5.8.35           5.8.35  6.6.0         ember-data > broccoli-babel-transpiler > babel-core                                                                        
fs-tree-diff                   0.3.1            0.3.1   0.4.4         ember-data > broccoli-babel-transpiler > broccoli-funnel > fs-tree-diff                                                    
minimatch                      2.0.10           2.0.10  3.0.0         ember-data > broccoli-babel-transpiler > broccoli-funnel > minimatch                                                       
minimatch                      2.0.10           2.0.10  3.0.0         ember-data > broccoli-babel-transpiler > broccoli-funnel > walk-sync > matcher-collection > minimatch                      
minimist                       0.0.8            0.0.8   1.2.0         ember-data > broccoli-babel-transpiler > broccoli-funnel > mkdirp > minimist                                               
mktemp                         0.3.5            0.3.5   0.4.0         ember-data > broccoli-babel-transpiler > broccoli-funnel > broccoli-plugin > quick-temp > mktemp                           
underscore.string              2.3.3            2.3.3   3.3.4         ember-data > broccoli-babel-transpiler > broccoli-funnel > broccoli-plugin > quick-temp > underscore.string                
rimraf                         2.2.8            2.2.8   2.5.2         ember-data > broccoli-babel-transpiler > broccoli-funnel > broccoli-plugin > quick-temp > rimraf                           
minimatch                      2.0.10           3.0.0   3.0.0         ember-data > broccoli-babel-transpiler > broccoli-funnel > rimraf > glob > minimatch                                       
fs-tree-diff                   0.3.1            0.3.1   0.4.4         ember-data > broccoli-babel-transpiler > broccoli-persistent-filter > fs-tree-diff                                         
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-data > broccoli-babel-transpiler > broccoli-persistent-filter > broccoli-kitchen-sink-helpers                        
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-data > broccoli-babel-transpiler > broccoli-persistent-filter > hash-for-dep > broccoli-kitchen-sink-helpers         
minimatch                      2.0.10           2.0.10  3.0.0         ember-data > broccoli-babel-transpiler > broccoli-persistent-filter > walk-sync > matcher-collection > minimatch           
minimist                       0.0.8            0.0.8   1.2.0         ember-data > broccoli-babel-transpiler > broccoli-persistent-filter > mkdirp > minimist                                    
mktemp                         0.3.5            0.3.5   0.4.0         ember-data > broccoli-babel-transpiler > broccoli-persistent-filter > broccoli-plugin > quick-temp > mktemp                
underscore.string              2.3.3            2.3.3   3.3.4         ember-data > broccoli-babel-transpiler > broccoli-persistent-filter > broccoli-plugin > quick-temp > underscore.string     
rimraf                         2.2.8            2.2.8   2.5.2         ember-data > broccoli-babel-transpiler > broccoli-persistent-filter > broccoli-plugin > quick-temp > rimraf                
rimraf                         2.2.8            2.2.8   2.5.2         ember-data > broccoli-merge-trees > broccoli-plugin > quick-temp > rimraf                                                  
underscore.string              2.3.3            2.3.3   3.3.4         ember-data > broccoli-merge-trees > broccoli-plugin > quick-temp > underscore.string                                       
mktemp                         0.3.5            0.3.5   0.4.0         ember-data > broccoli-merge-trees > broccoli-plugin > quick-temp > mktemp                                                  
rsvp                           3.0.21           3.0.21  3.2.1         ember-data > broccoli-file-creator > rsvp                                                                                  
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-data > broccoli-file-creator > broccoli-kitchen-sink-helpers                                                         
minimist                       0.0.8            0.0.8   1.2.0         ember-data > broccoli-file-creator > mkdirp > minimist                                                                     
rsvp                           3.0.21           3.2.1   3.2.1         ember-data > broccoli-file-creator > broccoli-writer > rsvp                                                                
rimraf                         2.2.8            2.2.8   2.5.2         ember-data > broccoli-file-creator > broccoli-writer > quick-temp > rimraf                                                 
mktemp                         0.3.5            0.3.5   0.4.0         ember-data > broccoli-file-creator > broccoli-writer > quick-temp > mktemp                                                 
underscore.string              2.3.3            2.3.3   3.3.4         ember-data > broccoli-file-creator > broccoli-writer > quick-temp > underscore.string                                      
rsvp                           3.0.21           3.2.1   3.2.1         ember-data > broccoli-file-creator > broccoli-plugin > promise-map-series > rsvp                                           
rimraf                         2.2.8            2.2.8   2.5.2         ember-data > broccoli-file-creator > broccoli-plugin > quick-temp > rimraf                                                 
mktemp                         0.3.5            0.3.5   0.4.0         ember-data > broccoli-file-creator > broccoli-plugin > quick-temp > mktemp                                                 
underscore.string              2.3.3            2.3.3   3.3.4         ember-data > broccoli-file-creator > broccoli-plugin > quick-temp > underscore.string                                      
semver                         4.3.6            4.3.6   5.1.0         ember-cli-qunit > ember-cli-version-checker > semver                                                                       
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-cli-qunit > broccoli-concat > broccoli-kitchen-sink-helpers                                                          
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-qunit > broccoli-concat > minimatch                                                                              
fast-sourcemap-concat          0.2.7            0.2.7   1.0.0         ember-cli-qunit > broccoli-concat > fast-sourcemap-concat                                                                  
minimist                       0.0.8            0.0.8   1.2.0         ember-cli-qunit > broccoli-concat > mkdirp > minimist                                                                      
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-cli-qunit > broccoli-concat > broccoli-caching-writer > broccoli-kitchen-sink-helpers                                
broccoli-plugin                1.1.0            1.1.0   1.2.1         ember-cli-qunit > broccoli-concat > broccoli-caching-writer > broccoli-plugin                                              
lodash.assign                  3.2.0            3.2.0   4.0.4         ember-cli-qunit > broccoli-concat > broccoli-caching-writer > lodash.assign                                                
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-qunit > broccoli-concat > broccoli-caching-writer > walk-sync > matcher-collection > minimatch                   
minimatch                      2.0.10           3.0.0   3.0.0         ember-cli-qunit > broccoli-concat > broccoli-caching-writer > rimraf > glob > minimatch                                    
rimraf                         2.2.8            2.2.8   2.5.2         ember-cli-qunit > broccoli-merge-trees > broccoli-plugin > quick-temp > rimraf                                             
underscore.string              2.3.3            2.3.3   3.3.4         ember-cli-qunit > broccoli-merge-trees > broccoli-plugin > quick-temp > underscore.string                                  
mktemp                         0.3.5            0.3.5   0.4.0         ember-cli-qunit > broccoli-merge-trees > broccoli-plugin > quick-temp > mktemp                                             
chalk                          0.4.0            0.4.0   1.1.1         ember-cli-qunit > broccoli-jshint > chalk                                                                                  
mkdirp                         0.4.2            0.4.2   0.5.1         ember-cli-qunit > broccoli-jshint > mkdirp                                                                                 
glob                           5.0.15           5.0.15  7.0.0         ember-cli-qunit > broccoli-jshint > findup-sync > glob                                                                     
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-qunit > broccoli-jshint > jshint > minimatch                                                                     
lodash                         3.7.0            3.7.0   4.5.1         ember-cli-qunit > broccoli-jshint > jshint > lodash                                                                        
strip-json-comments            1.0.4            1.0.4   2.0.1         ember-cli-qunit > broccoli-jshint > jshint > strip-json-comments                                                           
htmlparser2                    3.8.3            3.8.3   3.9.0         ember-cli-qunit > broccoli-jshint > jshint > htmlparser2                                                                   
shelljs                        0.3.0            0.3.0   0.6.0         ember-cli-qunit > broccoli-jshint > jshint > shelljs                                                                       
cli                            0.6.6            0.6.6   0.11.1        ember-cli-qunit > broccoli-jshint > jshint > cli                                                                           
date-now                       0.1.4            0.1.4   1.0.1         ember-cli-qunit > broccoli-jshint > jshint > console-browserify > date-now                                                 
fs-tree-diff                   0.3.1            0.3.1   0.4.4         ember-cli-qunit > broccoli-jshint > broccoli-persistent-filter > fs-tree-diff                                              
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-cli-qunit > broccoli-jshint > broccoli-persistent-filter > broccoli-kitchen-sink-helpers                             
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-cli-qunit > broccoli-jshint > broccoli-persistent-filter > hash-for-dep > broccoli-kitchen-sink-helpers              
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-qunit > broccoli-jshint > broccoli-persistent-filter > walk-sync > matcher-collection > minimatch                
minimist                       0.0.8            0.0.8   1.2.0         ember-cli-qunit > broccoli-jshint > broccoli-persistent-filter > mkdirp > minimist                                         
rimraf                         2.2.8            2.2.8   2.5.2         ember-cli-qunit > broccoli-jshint > broccoli-persistent-filter > broccoli-plugin > quick-temp > rimraf                     
underscore.string              2.3.3            2.3.3   3.3.4         ember-cli-qunit > broccoli-jshint > broccoli-persistent-filter > broccoli-plugin > quick-temp > underscore.string          
mktemp                         0.3.5            0.3.5   0.4.0         ember-cli-qunit > broccoli-jshint > broccoli-persistent-filter > broccoli-plugin > quick-temp > mktemp                     
clone                          0.2.0            0.2.0   1.0.2         ember-cli-qunit > broccoli-babel-transpiler > clone                                                                        
babel-core                     5.8.35           5.8.35  6.6.0         ember-cli-qunit > broccoli-babel-transpiler > babel-core                                                                   
fs-tree-diff                   0.3.1            0.3.1   0.4.4         ember-cli-qunit > broccoli-babel-transpiler > broccoli-funnel > fs-tree-diff                                               
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-qunit > broccoli-babel-transpiler > broccoli-funnel > minimatch                                                  
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-qunit > broccoli-babel-transpiler > broccoli-funnel > walk-sync > matcher-collection > minimatch                 
minimist                       0.0.8            0.0.8   1.2.0         ember-cli-qunit > broccoli-babel-transpiler > broccoli-funnel > mkdirp > minimist                                          
rimraf                         2.2.8            2.2.8   2.5.2         ember-cli-qunit > broccoli-babel-transpiler > broccoli-funnel > broccoli-plugin > quick-temp > rimraf                      
mktemp                         0.3.5            0.3.5   0.4.0         ember-cli-qunit > broccoli-babel-transpiler > broccoli-funnel > broccoli-plugin > quick-temp > mktemp                      
underscore.string              2.3.3            2.3.3   3.3.4         ember-cli-qunit > broccoli-babel-transpiler > broccoli-funnel > broccoli-plugin > quick-temp > underscore.string           
minimatch                      2.0.10           3.0.0   3.0.0         ember-cli-qunit > broccoli-babel-transpiler > broccoli-funnel > rimraf > glob > minimatch                                  
fs-tree-diff                   0.3.1            0.3.1   0.4.4         ember-cli-qunit > broccoli-babel-transpiler > broccoli-persistent-filter > fs-tree-diff                                    
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-cli-qunit > broccoli-babel-transpiler > broccoli-persistent-filter > broccoli-kitchen-sink-helpers                   
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-qunit > broccoli-babel-transpiler > broccoli-persistent-filter > walk-sync > matcher-collection > minimatch      
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-cli-qunit > broccoli-babel-transpiler > broccoli-persistent-filter > hash-for-dep > broccoli-kitchen-sink-helpers    
minimist                       0.0.8            0.0.8   1.2.0         ember-cli-qunit > broccoli-babel-transpiler > broccoli-persistent-filter > mkdirp > minimist                               
rimraf                         2.2.8            2.2.8   2.5.2         ember-cli-qunit > broccoli-babel-transpiler > broccoli-persistent-filter > broccoli-plugin > quick-temp > rimraf           
mktemp                         0.3.5            0.3.5   0.4.0         ember-cli-qunit > broccoli-babel-transpiler > broccoli-persistent-filter > broccoli-plugin > quick-temp > mktemp           
underscore.string              2.3.3            2.3.3   3.3.4         ember-cli-qunit > broccoli-babel-transpiler > broccoli-persistent-filter > broccoli-plugin > quick-temp > underscore.string
lodash.merge                   3.3.2            3.3.2   4.3.1         ember-cli-less > lodash.merge                                                                                              
semver                         4.3.6            4.3.6   5.1.0         ember-cli-less > ember-cli-version-checker > semver                                                                        
rimraf                         2.2.8            2.2.8   2.5.2         ember-cli-less > broccoli-merge-trees > broccoli-plugin > quick-temp > rimraf                                              
underscore.string              2.3.3            2.3.3   3.3.4         ember-cli-less > broccoli-merge-trees > broccoli-plugin > quick-temp > underscore.string                                   
mktemp                         0.3.5            0.3.5   0.4.0         ember-cli-less > broccoli-merge-trees > broccoli-plugin > quick-temp > mktemp                                              
lodash.merge                   3.3.2            3.3.2   4.3.1         ember-cli-less > broccoli-less-single > lodash.merge                                                                       
minimist                       0.0.8            0.0.8   1.2.0         ember-cli-less > broccoli-less-single > mkdirp > minimist                                                                  
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         ember-cli-less > broccoli-less-single > broccoli-caching-writer > broccoli-kitchen-sink-helpers                            
broccoli-plugin                1.1.0            1.1.0   1.2.1         ember-cli-less > broccoli-less-single > broccoli-caching-writer > broccoli-plugin                                          
lodash.assign                  3.2.0            3.2.0   4.0.4         ember-cli-less > broccoli-less-single > broccoli-caching-writer > lodash.assign                                            
minimatch                      2.0.10           2.0.10  3.0.0         ember-cli-less > broccoli-less-single > broccoli-caching-writer > walk-sync > matcher-collection > minimatch               
graceful-fs                    3.0.8            3.0.8   4.1.3         ember-cli-less > broccoli-less-single > less > graceful-fs                                                                 
promise                        6.1.0            6.1.0   7.1.1         ember-cli-less > broccoli-less-single > less > promise                                                                     
image-size                     0.3.5            0.3.5   0.4.0         ember-cli-less > broccoli-less-single > less > image-size                                                                  
source-map                     0.4.4            0.4.4   0.5.3         ember-cli-less > broccoli-less-single > less > source-map                                                                  
prr                            0.0.0            0.0.0   1.0.1         ember-cli-less > broccoli-less-single > less > errno > prr                                                                 
qs                             6.0.2            6.0.2   6.1.0         ember-cli-less > broccoli-less-single > less > request > qs                                                                
hawk                           3.1.3            3.1.3   4.1.2         ember-cli-less > broccoli-less-single > less > request > hawk                                                              
bl                             1.0.3            1.0.3   1.1.2         ember-cli-less > broccoli-less-single > less > request > bl                                                                
assert-plus                    0.2.0            0.2.0   1.0.0         ember-cli-less > broccoli-less-single > less > request > http-signature > assert-plus                                      
extsprintf                     1.0.2            1.0.2   1.3.0         ember-cli-less > broccoli-less-single > less > request > http-signature > jsprim > extsprintf                              
verror                         1.3.6            1.3.6   1.6.1         ember-cli-less > broccoli-less-single > less > request > http-signature > jsprim > verror                                  
assert-plus                    0.2.0            0.2.0   1.0.0         ember-cli-less > broccoli-less-single > less > request > http-signature > sshpk > assert-plus                              
supports-color                 2.0.0            2.0.0   3.1.2         ember-cli-less > broccoli-less-single > less > request > har-validator > chalk > supports-color                            
jsonpointer                    2.0.0            2.0.0   3.0.0         ember-cli-less > broccoli-less-single > less > request > har-validator > is-my-json-valid > jsonpointer                    
rsvp                           3.0.21           3.0.21  3.2.1         broccoli-asset-rev > rsvp                                                                                                  
minimatch                      2.0.10           2.0.10  3.0.0         broccoli-asset-rev > matcher-collection > minimatch                                                                        
rsvp                           3.0.21           3.2.1   3.2.1         broccoli-asset-rev > broccoli-filter > rsvp                                                                                
broccoli-kitchen-sink-helpers  0.2.9            0.2.9   0.3.1         broccoli-asset-rev > broccoli-filter > broccoli-kitchen-sink-helpers                                                       
rsvp                           3.0.21           3.2.1   3.2.1         broccoli-asset-rev > broccoli-filter > promise-map-series > rsvp                                                           
minimist                       0.0.8            0.0.8   1.2.0         broccoli-asset-rev > broccoli-filter > mkdirp > minimist                                                                   
rimraf                         2.2.8            2.2.8   2.5.2         broccoli-asset-rev > broccoli-filter > broccoli-plugin > quick-temp > rimraf                                               
underscore.string              2.3.3            2.3.3   3.3.4         broccoli-asset-rev > broccoli-filter > broccoli-plugin > quick-temp > underscore.string                                    
mktemp                         0.3.5            0.3.5   0.4.0         broccoli-asset-rev > broccoli-filter > broccoli-plugin > quick-temp > mktemp                                               
```

**Comment by sohara** [comment]
We are having this issue in our app as well. I've tried to weed out any older broccoli plugins by following ember-cli's performance guide. Currently we are getting builds looking basically like this (after writing an `hbs` file):

```
Build successful - 3114ms.

Slowest Trees                                 | Total
----------------------------------------------+---------------------
Babel                                         | 546ms
SourceMapConcat: Concat: App                  | 488ms
SassCompiler                                  | 478ms
StubGenerator                                 | 434ms
Funnel: Funnel: Pod Templates                 | 193ms

Slowest Trees (cumulative)                    | Total (avg)
----------------------------------------------+---------------------
Babel (25)                                    | 594ms (23 ms)
SourceMapConcat: Concat: App (1)              | 488ms
SassCompiler (1)                              | 478ms
StubGenerator (2)                             | 444ms (222 ms)
Funnel: Funnel: Pod Templates (1)             | 193ms
```

**Comment by aexmachina** [comment]
@stefanpenner may have some thoughts on this?

**Comment by olivierchatry** [comment]
Same here, my build jumped to 6 seconds or so, with SASS taking most of the time.

**Comment by stefanpenner** [comment]
> @stefanpenner may have some thoughts on this?

Most likely (one or both of the following):

* very large number of files in bower_components
* very large amount of sass compilation work

---

As usual, demo apps reproducing the above issues will help makes this more actionable. Although I am aware of at-least two issues, it would help to discover more.

**Comment by aexmachina** [comment]
Closing due to inactivity.

**Comment by CodyD604** [comment]
We noticed that any changes to files in our `includePaths` were causing sass recompilation in our apps as well. An optimization you can make to avoid invoking sass compilation on non-sass file changes is to use the `cacheInclude` option in your `sassOptions`. Ember-cli-sass passes the `cacheInclude` option to [broccoli-caching-writer](https://github.com/ember-cli/broccoli-caching-writer/tree/979abf92c83af7d625b1fd35c94b4e5f56668b18#documentation) which instructs the cached sass not only rebuild if it matches the pattern you pass to `cacheInclude`.

In our apps, we only care about recompiling sass on `.scss` changes, so we use the following configuration:
``` 
sassOptions: {
      ...,
      cacheInclude: [/\.scss$/]
}
```
Note that the pattern matching adds some time to your builds, but we found this was well-worth the trade-off. To rebuild on css/sass changes as well, you could use a regex [like the one included in 11.0.0](https://github.com/adopted-ember-addons/ember-cli-sass/commit/4d2c796539c6c766ee64ca0798f0c121bdad0d8f#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R20) (though this one does not apply to the `includePaths`).
